### PR TITLE
Fixes #25109 - add timeout for sync_task

### PR DESCRIPTION
**Pull request by iNecas**

### CODE REVIEW

File: lib/foreman_tasks.rb [hunk 1]
```diff
@@ -27,8 +27,15 @@ def self.trigger_task(async, action, *args, &block)
...
-              ForemanTasks::Task::DynflowTask.where(:external_id => id).first!
+              unless async
+                timeout = Setting['foreman_tasks_sync_task_timeout']
+                finished.wait(timeout)
+                task = ForemanTasks::Task::DynflowTask.where(:external_id => id).first
+                if task.nil? || task.pending?
+                  raise TimeoutError.new("The time waiting for task #{task.try(:id)} to finish exceeded the 'foreman_tasks_sync_task_timeout' (#{timeout}s)")
+                end
+                task
+              end
```
Comment (adamruzicka) [inline lib/foreman_tasks.rb]: If I'm reading this right, this will return `nil` if `async` is `true`

Comment (iNecas) [reply lib/foreman_tasks.rb]: good catch, updated

**Comment by iNecas** [comment]
Updated

**Comment by adamruzicka** [comment]
Thanks @iNecas !

**Pull request merged**